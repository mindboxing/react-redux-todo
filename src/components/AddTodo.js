import React, { useState } from 'react';
import { connect } from 'react-redux';
import { addTodo } from '../redux/actions';
import PropTypes from 'prop-types';


function AddTodo(props) {
  const [input, setInput] = useState("");

  const handleAddTodo = () => {
    props.addTodo(input);
    setInput("");
  }

  return (
    <div>
        <input
            onChange={e => setInput(e.target.value)}
            value={input}
        />
        <button className="add-todo" onClick={handleAddTodo}>
            Add Todo
        </button>                
    </div>
  );  
}

AddTodo.propTypes = {
    addTodo: PropTypes.func
}

export default connect(
    null, 
    { addTodo }
)(AddTodo);