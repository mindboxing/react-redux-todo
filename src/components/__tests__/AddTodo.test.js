import React from  "react";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import AddTodo from '../AddTodo';

const mockStore = configureStore([]);
function renderComponent(ui, state={}) {
  const store = mockStore(state);
  return [render((
      <Provider store={store}>
          {ui}
      </Provider>
  )), store];
}

describe('AddTodo', () => {

  afterAll(cleanup);

  test('should render', () => {
    renderComponent(<AddTodo/>);
    screen.debug();
  })
  
  test('should dispatch addTodo by clicking Add Todo button ', () => {
    const [_, store] = renderComponent(<AddTodo/>);
    const inoutNode = screen.getByText('', { selector: 'input' });
    fireEvent.change(inoutNode, { target: { value: 'Test Task' } });
    fireEvent.click(screen.getByText("Add Todo"));
    expect(store.getActions()).toContainEqual({payload: {content: "Test Task", id: 1}, type: "ADD_TODO"});
  })
})
