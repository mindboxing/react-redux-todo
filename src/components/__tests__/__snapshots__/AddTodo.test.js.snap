// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo should render 1`] = `
Array [
  Object {
    "asFragment": [Function],
    "baseElement": <body>
      <div>
        <div>
          <input
            value=""
          />
          <button
            class="add-todo"
          >
            Add Todo
          </button>
        </div>
      </div>
    </body>,
    "container": <div>
      <div>
        <input
          value=""
        />
        <button
          class="add-todo"
        >
          Add Todo
        </button>
      </div>
    </div>,
    "debug": [Function],
    "findAllByAltText": [Function],
    "findAllByDisplayValue": [Function],
    "findAllByLabelText": [Function],
    "findAllByPlaceholderText": [Function],
    "findAllByRole": [Function],
    "findAllByTestId": [Function],
    "findAllByText": [Function],
    "findAllByTitle": [Function],
    "findByAltText": [Function],
    "findByDisplayValue": [Function],
    "findByLabelText": [Function],
    "findByPlaceholderText": [Function],
    "findByRole": [Function],
    "findByTestId": [Function],
    "findByText": [Function],
    "findByTitle": [Function],
    "getAllByAltText": [Function],
    "getAllByDisplayValue": [Function],
    "getAllByLabelText": [Function],
    "getAllByPlaceholderText": [Function],
    "getAllByRole": [Function],
    "getAllByTestId": [Function],
    "getAllByText": [Function],
    "getAllByTitle": [Function],
    "getByAltText": [Function],
    "getByDisplayValue": [Function],
    "getByLabelText": [Function],
    "getByPlaceholderText": [Function],
    "getByRole": [Function],
    "getByTestId": [Function],
    "getByText": [Function],
    "getByTitle": [Function],
    "queryAllByAltText": [Function],
    "queryAllByDisplayValue": [Function],
    "queryAllByLabelText": [Function],
    "queryAllByPlaceholderText": [Function],
    "queryAllByRole": [Function],
    "queryAllByTestId": [Function],
    "queryAllByText": [Function],
    "queryAllByTitle": [Function],
    "queryByAltText": [Function],
    "queryByDisplayValue": [Function],
    "queryByLabelText": [Function],
    "queryByPlaceholderText": [Function],
    "queryByRole": [Function],
    "queryByTestId": [Function],
    "queryByText": [Function],
    "queryByTitle": [Function],
    "rerender": [Function],
    "unmount": [Function],
  },
  Object {
    "clearActions": [Function],
    "dispatch": [Function],
    "getActions": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
]
`;
